R/R/step.R

```diff
@@ -31,6 +31,9 @@ step <- function(flow, ..., step, r_function = NULL, foreach = NULL, join = FALS
   }
   if (!is.null(r_function)) {
     function_name <- as.character(substitute(r_function))
+    if (length(function_name) > 1) { # likely an anonymous function
+      function_name <- paste0(step, "_function")
```

Comment:
Thanks for the improvement! Can we add hash of the function to this function name as well? Just in case the user may have other non-anonymous function with the same name @mdneuzerling

Reply:
Sorry for the delay, @jasonge27. Base R doesn't have any hashing capabilities, so I would need to add a dependency to the `digest` package. Would this be okay? I also considered appending the location of the function in memory instead, but even this would require an additional dependency (`pryr` or `lobstr`).

Reply:
thanks for the PR @mdneuzerling!

Would using the python builtin hash function save the dependency here? 

```
library(reticulate)
builtins <- import_builtins()
py_hash <- function(fun) {
  builtins$hash(paste(deparse(fun), collapse = ""))
}


> sumfun <- function(x, y) { x + y}
> py_hash(sumfun)
[1] -61687620
>
> sumfun1 <- function(x, y) {x + y + y}
> py_hash(sumfun1)
[1] -504669628
```

Reply:
Thanks @bcgalvin. I've got that working, although [the inbuilt Python hash function salts the inputs and so the hash will change between sessions](https://docs.python.org/3/reference/datamodel.html#object.__hash__). If this behaviour is undesirable, I think a dependency on either R's `digest` package or Python's `hashlib` module is needed.

Reply:
Hey thank you David for the updated PR! @mdneuzerling (not sure why ..Github is not letting me mention you). I discussed with the team. We believe it's better to use `digest` since :
1.  as you've mentioned the native python hash is not very stable (overflow & changing between sessions) 
2. The `digest` pkg is maintained by Dirk and does not introduce additional dependency (`utils` pkg is already required by `reticulate`) . 

I saw that you've gone via the python path. Sorry for the late reply. Do you have time to update the PR & use `digest`?

Reply:
Not a problem. I"m not an expert on hashing algorithms, but I've heard good things about spookyhash for non-cryptographic purposes, so I went with that.

``` r
#> flow <- metaflow("TestFlow") %>%
#     step(
#       step = "anonymous",
#       r_function = function(step) step$x <- 3 
#     )
#> names(flow$get_functions())
"anonymous_function_7542958bbcd6fa74bdf17e888b956426"
```

Reply:
Thank you @mdneuzerling for the quick turnaround. Merging the PR.